sdk/modelsrepository/Azure.Iot.ModelsRepository/tests/ModelQueryTests.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Iot.ModelsRepository.Tests
+{
+    public class ModelQueryTests
+    {
+        private readonly string _modelTemplate = @"{{
+            {0}
+            ""@type"": ""Interface"",
+            ""displayName"": ""Phone"",
+            {1}
+            {2}
+            ""@context"": ""dtmi:dtdl:context;2""
+        }}";
+
+        [TestCase("\"@id\": \"dtmi:com:example:thermostat;1\",", "dtmi:com:example:thermostat;1")]
+        [TestCase("\"@id\": \"\",", "")]
+        [TestCase("", "")]
+        public void GetId(string formatId, string expectedId)
+        {
+            string modelContent = string.Format(_modelTemplate, formatId, "", "");
+            ModelQuery query = new ModelQuery(modelContent);
+            Assert.AreEqual(query.GetId(), expectedId);
+        }
+
+        [TestCase(
```

Comment:
Not in live with these massive attribute parameters. I think these are good for simple parameters, for for big json payloads, we should refer to a resx file, constants, or files on disk.

That may mean we can't use data-driven test cases, but so be it.